go-controller/pkg/ovn/gress_policy.go

```diff
@@ -459,8 +459,9 @@ func getSvcVips(nbClient client.Client, service *v1.Service) []net.IP {
 		}
 	}
 	if util.ServiceTypeHasClusterIP(service) {
-		if util.IsClusterIPSet(service) {
-			ip := net.ParseIP(service.Spec.ClusterIP)
+		ipStrs := util.GetClusterIPs(service)
+		for _, ipStr := range ipStrs {
+			ip := net.ParseIP(ipStr)
 			if ip == nil {
 				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
```

Comment:
oops this is missing a continue, right?

Reply:
Yes, the original code had the issue, too, though:

Original:
~~~
		if util.IsClusterIPSet(service) {
			ip := net.ParseIP(service.Spec.ClusterIP)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
			}
			ips = append(ips, ip)
		}
~~~

Should probably have been:
~~~
		if util.IsClusterIPSet(service) {
			ip := net.ParseIP(service.Spec.ClusterIP)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
			} else {
			        ips = append(ips, ip)
                        }
		}
~~~

New version:
~~~
		for _, ipStr := range ipStrs {
			ip := net.ParseIP(ipStr)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
			}
			ips = append(ips, ip)
		}
~~~

Should probably be:
~~~
		for _, ipStr := range ipStrs {
			ip := net.ParseIP(ipStr)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
                                continue
			}
			ips = append(ips, ip)
		}
~~~

Reply:
I just fixed this for Tim and pushed.